File: app/src/main/java/com/lany/fivechess/activity/MainActivity.java
package com.lany.fivechess.activity;

import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AlertDialog;
import android.view.View;

import com.lany.fivechess.R;


public class MainActivity extends BaseActivity {
    @Override
    protected boolean hasBackButton() {
        return false;
    }

    @Override
    protected int getLayoutId() {
        return R.layout.activity_main;
    }

    @Override
    protected void init(Bundle savedInstanceState) {
        findViewById(R.id.new_game).setOnClickListener(new View.OnClickListener() {

            @Override
            public void onClick(View v) {
                startActivity(new Intent(MainActivity.this,
                        RobotGameActivity.class));
            }
        });

        findViewById(R.id.fight).setOnClickListener(new View.OnClickListener() {

            @Override
            public void onClick(View v) {
                startActivity(new Intent(MainActivity.this,
                        PersonGameActivity.class));
            }
        });

        findViewById(R.id.conn_fight).setOnClickListener(new View.OnClickListener() {

            @Override
            public void onClick(View v) {
                startActivity(new Intent(MainActivity.this,
                        ConnectionActivity.class));
            }
        });

        findViewById(R.id.conn_about).setOnClickListener(new View.OnClickListener() {

            @Override
            public void onClick(View v) {
                showAboutDialog();
            }
        });
    }

    private void showAboutDialog() {
        AlertDialog.Builder b = new AlertDialog.Builder(this);
        b.setTitle(R.string.about);
        b.setMessage("欢迎访问源代码\nhttps://github.com/lany192/FiveChess");
        b.setPositiveButton(R.string.ok,
                new DialogInterface.OnClickListener() {

                    @Override
                    public void onClick(DialogInterface dialog, int which) {

                    }
                });
        b.show();
    }

}

File: app/src/main/java/com/lany/fivechess/activity/MainActivity.kt
package com.lany.fivechess.activity

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AlertDialog
import com.lany.fivechess.R
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : BaseActivity() {

    override fun hasBackButton(): Boolean {
        return false
    }

    override fun getLayoutId(): Int {
        return R.layout.activity_main
    }

    override fun init(savedInstanceState: Bundle?) {
        new_game.setOnClickListener {
            startActivity(Intent(this@MainActivity, RobotGameActivity::class.java))
        }
        fight.setOnClickListener {
            startActivity(Intent(this@MainActivity, PersonGameActivity::class.java))
        }
        conn_fight.setOnClickListener {
            startActivity(Intent(this@MainActivity, ConnectionActivity::class.java))
        }
        conn_about.setOnClickListener {
            val b = AlertDialog.Builder(this@MainActivity)
            b.setTitle(R.string.about)
            b.setMessage("欢迎访问源代码\nhttps://github.com/lany192/FiveChess")
            b.setPositiveButton(R.string.ok) { dialog, which -> }
            b.show()
        }
    }
}
